image-patching/stopgap-script/nuage_overcloud_full_patch.py

```diff
@@ -164,69 +152,57 @@ def uninstall_packages(image):
         image, VIRT_CUSTOMIZE_MEMSIZE))
 
 
-#####
-# Function to add files based on the version
-#####
-
-def add_files(image, version, workingDir):
-    version = int(version)
-    if version == 13:
-        cmds_run(['cat <<EOT > version_13 \n'
-                  'cp /etc/puppet/modules/nuage/manifests/13_files'
-                  '/neutron_init.pp '
-                  '/etc/puppet/modules/neutron/manifests/init.pp \n'
-                  'cp /etc/puppet/modules/nuage/manifests/13_files'
-                  '/conductor.pp '
-                  '/etc/puppet/modules/ironic/manifests/conductor.pp \n'
-                  'EOT'])
-        virt_customize(
-            '"mkdir -p /etc/puppet/modules/nuage/manifests/13_files" '
-            '-a %s --memsize %s --selinux-relabel' % (
-                image, VIRT_CUSTOMIZE_MEMSIZE))
-        virt_copy(
-            '%s %s/13_files/* '
-            '/etc/puppet/modules/nuage/manifests/13_files' % (
-            image, workingDir))
-        virt_customize_run(
-            'version_13 -a %s --memsize %s --selinux-relabel' % (
-            image, VIRT_CUSTOMIZE_MEMSIZE))
-
-        cmds_run(['rm -f version_13'])
-
-
 #####
 # Function to install Nuage packages that are required
 #####
 
 def install_packages(image):
-    cmds_run(['cat <<EOT > nuage_packages \n'
-              'yum install %s -y \n'
-              'yum install %s -y \n'
-              'yum install %s -y \n'
-              'EOT' % (
-              NUAGE_DEPENDENCIES, NUAGE_PACKAGES, NUAGE_VRS_PACKAGE)])
+    cmds_run([
+        'cat <<EOT > nuage_packages \n'
+        'yum install {0} -y \n'
+        'yum install {1} -y \n'
+        'yum install {2} -y \n'
+        'missing_packages="" \n'
+        'for package in {0} {1} {2} \n'
+        'do \n'
+        '    echo "Checking if \$package is present or not ?" \n'
+        '    if rpm -qa | grep \$package  2>&1 > /dev/null; then \n'
+        '      echo " \$package exists" \n'
+        '    else \n'
+        '        echo " \$package does not exits" \n'
+        '        missing_packages+=" \$package" \n'
+        '    fi \n'
+        'done \n'
+        'if [ -z "\$missing_packages" ] \n'
+        'then\n'
+        '    echo "Verification Completed!!" \n'
+        '    exit 0 \n'
+        'else \n'
+        '    echo "Verification failed with missing packages '
+        '\$missing_packages" \n'
+        '    exit 1 \n'
+        'fi\n'
+        'EOT'.format(NUAGE_DEPENDENCIES, NUAGE_PACKAGES, NUAGE_VRS_PACKAGE)
```

Comment:
NUAGE_PACKAGES has nuage-metadata-agent, which requires nuage-openvswitch and according to this fucntion, nuage-openvswitch is installing later in  NUAGE_VRS_PACKAGE